src/ddceler/DDcelerTMI.hh

```diff
@@ -0,0 +1,67 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file ddceler/DDcelerTMI.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <vector>
+#include <CeleritasG4.hh>
+#include <DDG4/Geant4Action.h>
+#include <DDG4/Geant4PhysicsList.h>
+#include <G4VModularPhysicsList.hh>
+
+using Geant4Context = dd4hep::sim::Geant4Context;
+using Geant4Action = dd4hep::sim::Geant4Action;
+using Geant4PhysicsList = dd4hep::sim::Geant4PhysicsList;
+
+namespace dd4hep
+{
+namespace sim
+{
+//---------------------------------------------------------------------------//
+/*!
+ * DDG4 action plugin for Celeritas tracking manager integration (TMI).
+ */
+class DDcelerTMI : public Geant4PhysicsList
+{
+  protected:
+    // Define standard assignments and constructors
+    DDG4_DEFINE_ACTION_CONSTRUCTORS(DDcelerTMI);
+
+    int m_maxNumTracks;
+    int m_initCapacity;
```

Comment:
These should be default initialized. Maybe use -1, and then VALIDATE that they're greater than zero when you first set up, to ensure that the user has set them to appropriate values?